### PR TITLE
Fill in Security and Privacy Considerations sections

### PR DESCRIPTION
**Pull request by jcjones**
These are all the considerations from my notes, as well as a few more that I've dreamed up.

The first commit also updates the README a bit.

### CODE REVIEW

File: draft-stark-expect-ct.md [hunk 1]
```diff
@@ -508,12 +508,61 @@ UA SHOULD report the failure as follows:
 
 # Security Considerations
 
+When UAs support the Expect-CT header, it becomes a potential vector for hostile
+header attacks against site owners. If a site owner uses a certificate issued by
+a certificate authority which does not embed SCTs nor serve SCTs via OCSP, a
```
Comment (estark37) [inline draft-stark-expect-ct.md]: nor serve SCTs via TLS extension

File: draft-stark-expect-ct.md [hunk 2]
```diff
@@ -508,12 +508,61 @@ UA SHOULD report the failure as follows:
...
+reach its `max-age`. User agents may choose to implement mechanisms for users to
+cure this situation, as noted in {{usability-considerations}}.
+
 ## Maximum max-age
 
-1 year?
+There is a security trade-off in that low maximum values provide a narrow window
+of protection for users that visit the Known Expect-CT Host only infrequently,
+while high maximum values might result in a denial of service to a UA in the
+event of a hostile header attack.
```
Comment (estark37) [inline draft-stark-expect-ct.md]: or site owner error

File: draft-stark-expect-ct.md [hunk 3]
```diff
@@ -508,12 +508,61 @@ UA SHOULD report the failure as follows:
...
+
+There is probably no ideal maximum for the `max-age` directive. Since Expect-CT
+is primarily a policy-expansion and investigation technology rather than an
+end- user protection, a value on the order of 30 days (2,592,000 seconds) may be
```
Comment (estark37) [inline draft-stark-expect-ct.md]: nit: there's an extra space in there after "end-"

File: draft-stark-expect-ct.md [hunk 4]
```diff
@@ -508,12 +508,61 @@ UA SHOULD report the failure as follows:
...
+Expect-CT can be used to infer what Certificate Transparency policy is in use,
+by attempting to retrieve specially-configured websites which pass one user
+agents' policies but not another's. Note that this consideration is true of UAs
+which enforce CT policies without Expect-CT as well.
+
+Additionally, reports submitted to the `report-uri` could reveal information to
+a third party about which webpage is being accessed and by which IP address, by
+using individual `report-uri` values for individually-tracked pages. This
+information could be leaked even if client-side scripting were disabled.
+
```
Comment (estark37) [inline draft-stark-expect-ct.md:567]: There's also the HPKP super-cookie consideration in https://tools.ietf.org/html/rfc7469#section-5. I think the same thing applies here, do you agree?

**Review by estark37 — COMMENTED** [review]
Thanks @jcjones! Left a few comments inline, otherwise looks good to me.

**Comment by jcjones** [comment]
I've reviewed all of your comments, and thought through privacy considerations some more, taking more ideas from RFC7469. One I didn't resolve was about the super-cookie consideration: I'm not (yet) able to convince myself of how that would work, so I'm not sure how to write it up.

The HPKP super-cookie is dependent upon two things:
1) The ease of setting pins that aren't checked at "note time"
2) {less important} Cascading set pins to subdomains

Expect-CT doesn't have an equivalent to setting unchecked pins, and nor does it have `includeSubDomains`. So it seems to be to produce a super-cookie scenario one would need to:

1) Set up `X` tracking websites, each in a good Expect-CT state and emitting Expect-CT headers
2) Compel clients to Note a unique subset of those `X` tracking websites, `Y` where `Y ⊂ X`.
3) Reconfigure a subset `Z` of those `X` tracking websites to violate CT policy where `Z ⊂ X`.
4) Compel clients to connect to all of those `X` tracking websites and use the resulting failure list to derive the client's identity.

This seems to be much harder than providing unique `report-uri` values to each client and prompting a failure... but I'm probably missing something obvious!

I'll be happy to talk through the attack a bit more and take another try at it, maybe in a different PR.

**Comment by estark37** [comment]
Yeah, I was thinking about the unique-subset-of-noted-subdomains thing (e.g. http://www.leviathansecurity.com/blog/the-double-edged-sword-of-hsts-persistence-and-privacy/), which I thought was in both HPKP and HSTS RFCs, but I guess I was mistaken because I don't see it in either of them!

I still think it might be worth including but agree that it can be done in a different PR.

**Comment by estark37** [comment]
This PR lgtm, just have to figure out what's going on with the circleci check...

**Comment by estark37** [comment]
@bifurcation do you have any idea what's up with the circleci check?

**Comment by bifurcation** [comment]
@estark37 I don't see a [CircleCI build](https://circleci.com/gh/bifurcation/expect-ct) for it (??)  The main thing CircleCI checks is that the document builds, so if you can do that locally, it's probably safe to just merge.

**Comment by estark37** [comment]
@bifurcation I don't seem to be able to merge (either manually or in the github UI) without the CircleCI check passing.

**Comment by jcjones** [comment]
I've committed an amend and forced an update; it looks like CircleCI passed this time. @estark37

**Pull request merged**